### PR TITLE
Update Query Group API descriptions

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Updates the Query Group API descriptions to add automation to Query Group API reference. 

@natebower should approve these descriptions before merging.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/namespaces/wlm.yaml [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
+  description: The Query Group Lifecycle APIs creates, updates, and deletes Workload Management query groups.
   version: 1.0.0
 paths:
   /_wlm/query_group:
     get:
       operationId: wlm.get_query_group.0
       x-operation-group: wlm.get_query_group
       x-version-added: '2.17'
-      description: Gets the specified QueryGroup or get all if no name is provided.
+      description: Retrieves the specified Query Group.
```
Comment (Naarcha-AWS) [inline spec/namespaces/wlm.yaml:12]:
```suggestion
      description: Retrieves the specified query group. If no query group is specified, retrieves all query groups in the cluster.
```

File: spec/schemas/wlm._common.yaml [hunk 2]
```diff
@@ -10,16 +10,21 @@ components:
...
         resiliency_mode:
+          description: The resiliency mode of the query group.
           type: string
           enum: [enforced, monitor,soft]
         updated_at:
+          description: The time the query group was last updated.
           type: integer
           format: int64
         resource_limits:
+          description: The resource limits for query requests made in the query group.
```
Comment (natebower) [inline spec/schemas/wlm._common.yaml:27]:
```suggestion
          description: The resource limits for query requests in the query group.
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS Please see my comment and changes and let me know if you have any questions. Thanks!

**Comment by Naarcha-AWS** [comment]
@dblock: Editorial review complete. Let me know if I need to make any additional tweaks.

**Review by dblock — APPROVED** [review]

**Pull request merged**